### DISCUSSION TITLE
Missing external console option in run configuration, why?

### DISCUSSION THREAD

**Discussion by PerfectlyFineCode** [intellij-rust/intellij-rust#9717] [Other]
Hi!

I've recently started to make a TUI application, while an internal console is nice to have. It can be a bit of a pain because it can break TUI applications. Example being https://github.com/intellij-rust/intellij-rust/issues/7602

How come such an option does not exist?
Is it just not a very requested feature?

**Comment by mchernyavsky** [comment]
Doesn't `Emulate terminal in output console` option work for you? (https://github.com/intellij-rust/intellij-rust/issues/7602#issuecomment-1325842318)

**Reply by PerfectlyFineCode** [reply]
It does not, no.

It breaks input, as well when debugging it doesn't work at all.

**Reply by PerfectlyFineCode** [reply]
Rust is one of the only plugins that does not allow for external consoles.
Which is pretty annoying, is there a limitation or something?

**Reply by vlad20012** [reply]
There isn't an external console for Java/Kotlin (which we usually use) so I even didn't know such option exist somewhere or it is even possible. In which language plugins such an option exists?

**Reply by PerfectlyFineCode** [reply]
> There isn't an external console for Java/Kotlin (which we usually use) so I even didn't know such option exist somewhere or it is even possible. In which language plugins such an option exists?

Afaik C / C++, C#

**Comment by PerfectlyFineCode** [comment]
Example:
![image](https://user-images.githubusercontent.com/70201838/203771094-6c1e364a-db54-445a-ae3e-4a0b6f3fa1ca.png)

**Library used**
Anathema (https://crates.io/crates/anathema) which is using the crossterm (https://crates.io/crates/crossterm) crate.

**Comment by PerfectlyFineCode** [comment]
No updates so far.
I'm a little bit disappointed that people seem to be fine with the awful built-in terminal.
